### ISSUE TITLE
How to use buck2 modes ?

### ISSUE THREAD

**Issue by MarcoBonino** [facebook/buck2#779]
Hello!

I have a small example here ([buck2_example](https://github.com/MarcoBonino/buck2_example)) where I try to use buck2 mode during build. 
I run it using `buck2 run @mode/opt //:main`, but I'm getting this error:
```
thread 'main' has overflowed its stack
fatal runtime error: stack overflow
[1]    10198 abort      buck2 run @mode/opt //:main
```

Could you help ?
For more context what I would like to achieve is to be able to invoke buck2 build with different compilers and compiler flags (and probably also different cpu architectures). Could you advice on the best practices for it ? From the official documentation is not quite described or at least I could not found it.

Thanks!

**Comment by steveklabnik** [comment]
This is just a wild guess, but maybe this line shouldn't be here? https://github.com/MarcoBonino/buck2_example/blob/main/mode/opt#L2 isn't that like, recursively including itself? that would explain the stack overflow.

**Comment by Nero5023** [comment]
yeah, that line should not be there. So close this issue first. Feel free to reopen it if have any further question
